src/Maestro/Microsoft.DotNet.Maestro.Tasks.Tests/GetBuildManifestMetadataTests.cs

```diff
@@ -0,0 +1,115 @@
+using System;
+using System.Linq;
+using System.Threading;
+using FluentAssertions;
+using NUnit.Framework;
+
+namespace Microsoft.DotNet.Maestro.Tasks.Tests
+{
+    [TestFixture]
+    public class GetBuildManifestMetadataTests
+    {
+        private PushMetadataToBuildAssetRegistry pushMetadata;
+        private string testManifestFolderPath = @".\TestManifests\"; 
+
+        public GetBuildManifestMetadataTests()
+        {
+            pushMetadata = new PushMetadataToBuildAssetRegistry();
+        }
+        [Test]
+        public void EmptyManifestFolderPath()
+        {
+            Action act = () => pushMetadata.GetBuildManifestsMetadata("", new CancellationToken());
```

Comment:
The fact that all of these tests _do_ pass a cancellation token sort of makes we want tests for what happens if the token gets cancelled.  That does basically double the number of tests though...

Reply:
Could parameterize the test and see what happens :)

Reply:
They should, presumably, all just throw a cancellation exception without any of the side effects happening.

Reply:
It throws before parsing the next set of files if it's cancelled, so it just tosses anything that's already been done and stops doing things. Nothing in this method is assigning to class variables (it returns a Tuple of the things it created), so it shouldn't have any effects. I don't know that it makes sense to double the number of tests for this (since they should all do the same thing), but I'll look at adding one to make sure it's cancelling correctly.

Reply:
Yeah, probably not interesting if there aren't any observable side effects.  It would only be useful if a method had some sort of side effect, and the cancelled thing shouldn't have those side effects.  Proving that cancelled throws the cancelled isn't really interesting.  You could make it more interesting by making one of the underlying things block forever and make sure it cancels "fast". But that's a huge amount of work, especially when dependencies aren't set up well.

Reply:
Nothing interesting happens with the token, so skipping this change.